File: 3.c
#include <stdio.h>

int main() {
    double sum=1;
    double presum=0;
    double condition=1;
    double d,summ;
    int n,m;
    printf("Integer input:");
    scanf("%d", &n);
    if(n>0){
        for(int i=1;sum-presum>0.001 && condition==1;){
            if (i==n){condition=0;}
            else{
                i=i+1;
                d=i;
                presum=sum;
                sum=sum+1/d;
            }
        }
        printf("%lf\n", sum);
    }
    else if(n==0){printf("Sum is 1 or not defined as per the # QUESTION");}
    else{
        m=-n;
        for(int i=1;sum-presum>0.001 && condition==1;){
            if (i==m){condition=0;}
            else{
                i=i+1;
                d=i;
                presum=sum;
                sum=sum+1/d;
            }
        }
        summ=-sum;
        printf("%lf\n", summ);
    }
    return 0;
}

File: 3.py
num = input('Please type the number:')
try:
    n = int(num)
    condition = True
    count = 1
    sum = 1
    cs = 0
    if n > 0:
        while sum-cs > 0.001 and condition:
            if count == n:
                condition = False
            else:
                count = count+1
                cs = sum
                sum = sum+1/count
        print("Sum:", sum, "Number of terms:", count)
    elif n == 0:
        print("Sum is 1 or not defined as per the # QUESTION")
    else:
        nn = -1*n
        while sum-cs > 0.001 and condition:
            if count == nn:
                condition = False
            else:
                count = count+1
                cs = sum
                sum = sum+1/count
        summ = -1*sum
        print("Sum:", summ, "Number of terms:", count)
except ValueError:
    print('Type an integer!')
